main.sh

```diff
@@ -91,12 +92,27 @@ trap cleanup_before_exit EXIT
 
 # Translate usage string -> getopts arguments, and set $arg_<flag> defaults
 while read line; do
-  opt="$(echo "${line}" |awk '{print $1}' |sed -e 's#^-##')"
+  # fetch single character version of option sting
+  opt="$(echo "${line}" |awk '{print $1}' |sed -e 's#^-\(.\)#\1#')"
+
+  # fetch long version if present
+  long_opt="$(echo "${line}" |awk 'match($2,"^--.*",a){print substr(a[0],3)}')"
+  # map long name back to short name
+  varname="short_opt_${long_opt}"
+  eval "${varname}=\"${opt}\""
+  # map short name to long name ... I don't think this is needed
+  varname="long_opt_${opt:0:1}"
+  eval "${varname}=\"${long_opt}\""
```

Comment:
I will probably remove this from the PR, since it is unused, but wanted to demonstrate how to map going the other direction